greedy_algorithms/dijkstra_spf.cpp

```diff
@@ -94,31 +93,32 @@ void Dijkstra(Graph graph, int src) {
 int main() {
     int V, E, gsrc;
     int src, dst, weight;
-    cout << "Enter number of vertices: ";
-    cin >> V;
-    cout << "Enter number of edges: ";
-    cin >> E;
+    std::cout << "Enter number of vertices: ";
+    std::cin >> V;
+    std::cout << "Enter number of edges: ";
+    std::cin >> E;
     Graph G(V);
     for (int i = 0; i < E; i++) {
-        cout << "\nEdge " << i + 1 << "\nEnter source: ";
-        cin >> src;
-        cout << "Enter destination: ";
-        cin >> dst;
-        cout << "Enter weight: ";
-        cin >> weight;
+        std::cout << "\nEdge " << i + 1 << "\nEnter source: ";
+        std::cin >> src;
+        std::cout << "Enter destination: ";
+        std::cin >> dst;
+        std::cout << "Enter weight: ";
+        std::cin >> weight;
 
         // makes sure source and destionation are in the proper bounds.
         if (src >= 0 && src < V && dst >= 0 && dst < V) {
             G.addEdge(src, dst, weight);
         } else {
-            cout << "source and/or destination out of bounds" << endl;
+            std::cout << "source and/or destination out of bounds" << std::endl;
             i--;
             continue;
         }
     }
-    cout << "\nEnter source:";
-    cin >> gsrc;
+    std::cout << "\nEnter source:";
+    std::cin >> gsrc;
     Dijkstra(G, gsrc);
+    delete[] G.edges;
```

Comment:
This delete is incorrect. 
Please add a destructor to the Graph class that will deallocate the memory.